2020 has generally been a long year for people all over the world. Due to the pandemic, many people were forced to stay at home for extended periods of time which consequently led to a rise in gaming. As we near the end of the year, the two biggest console developers in Microsoft and Sony are releasing  

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

C:\ProgramData\Anaconda3\lib\site-packages\statsmodels\tools\_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [2]:
play_store_raw_df = pd.read_csv(r"C:\Users\yadelaja\Downloads\googleplaystore\googleplaystore.csv")
play_store_raw_df.loc[50:55]

App           Category  \
50                               Real Tractor Farming  AUTO_AND_VEHICLES   
51                    Ultimate F1 Racing Championship  AUTO_AND_VEHICLES   
52                      Used Cars and Trucks for Sale  AUTO_AND_VEHICLES   
53                           American Muscle Car Race  AUTO_AND_VEHICLES   
54     Offroad Oil Tanker Driver Transport Truck 2019  AUTO_AND_VEHICLES   
55  Tickets SDA 2018 and Exam from the State Traff...  AUTO_AND_VEHICLES   

    Rating Reviews                Size    Installs  Type Price Content Rating  \
50     4.0    1598                 56M  1,000,000+  Free     0       Everyone   
51     3.8     284                 57M    100,000+  Free     0       Everyone   
52     4.6   17057  Varies with device  1,000,000+  Free     0       Everyone   
53     3.9     129                 35M    100,000+  Free     0       Everyone   
54     4.3     542                 33M    100,000+  Free     0       Everyone   
55     4.9   10479                 33M    100,000+  Free     0       Everyone   

             Genres   Last Updated         Current Ver         Android Ver  
50  Auto & Vehicles  July 26, 2018                11.0          4.1 and up  
51  Auto & Vehicles  July 26, 2018                 3.0          4.1 and up  
52  Auto & Vehicles  July 30, 2018  Varies with device  Varies with device  
53  Auto & Vehicles  July 26, 2018                 3.0          4.1 and up  
54  Auto & Vehicles  July 31, 2018                 4.0          4.1 and up  
55  Auto & Vehicles  July 18, 2018               1.7.1          4.0 and up

Looking at the list of columns, I am uninterested in the size of the apps as well as the last 4 columns. I used the df.drop() method to get rid of those columns.

In [3]:
column_names_raw = play_store_raw_df.columns

rename_columns = {}
for column in column_names_raw:
  rename_columns[column] = column.replace(' ','_').lower()

play_store_raw_df.rename(columns=rename_columns, inplace=True)
play_store_raw_df

app             category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       rating reviews                size     installs  type price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      content_rating                     genres      last_updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              current_ver         android_ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[10841 rows x 13 columns]

In [4]:
column_names = play_store_raw_df.columns
column_names
drop_columns = [column_names[i] for i in [4, 9, 10, 11, 12]]
play_store_raw_df.drop(drop_columns, inplace=True, axis=1)
play_store_raw_df

app             category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       rating reviews     installs  type price content_rating  
0         4.1     159      10,000+  Free     0       Everyone  
1         3.9     967     500,000+  Free     0       Everyone  
2         4.7   87510   5,000,000+  Free     0       Everyone  
3         4.5  215644  50,000,000+  Free     0           Teen  
4         4.3     967     100,000+  Free     0       Everyone  
...       ...     ...          ...   ...   ...            ...  
10836     4.5      38       5,000+  Free     0       Everyone  
10837     5.0       4         100+  Free     0       Everyone  
10838     NaN       3       1,000+  Free     0       Everyone  
10839     4.5     114       1,000+  Free     0     Mature 17+  
10840     4.5  398307  10,000,000+  Free     0       Everyone  

[10841 rows x 8 columns]

Now that we have loaded the data into a pandas dataframe, we can explore some introductory information about our dataset. From the columns in the dataframe, we can see that this dataset contains information about the app name, category, rating, number of reviews, install count etc. To get the size of the dataframe, we use the `df.shape` method. `df.info()` is also really useful to get a quick overview of the datatypes of the columns as well as the non-null counts which allows us to identify potentially incorrect/missing data fields. 

In [5]:
play_store_raw_df.shape

(10841, 8)

In [6]:
play_store_raw_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   app             10841 non-null  object 
 1   category        10841 non-null  object 
 2   rating          9367 non-null   float64
 3   reviews         10841 non-null  object 
 4   installs        10841 non-null  object 
 5   type            10840 non-null  object 
 6   price           10841 non-null  object 
 7   content_rating  10840 non-null  object 
dtypes: float64(1), object(7)
memory usage: 677.7+ KB


Using `df.shape` we can identify a couple of issues with the dataset:

*   Not all the columns have 10840 entries which indicates missing/incorrect data
*   columns like reviews, size, installs and price should be of the type 'int64/float64', however they are of type 'object'. This is a problem as we cannot perform numerical analysis on type 'object' data.

In [7]:
play_store_raw_df['reviews'] = play_store_raw_df['reviews'].str.replace('.0M','0')

play_store_raw_df['installs'] = play_store_raw_df['installs'].str.replace('+','')
play_store_raw_df['installs'] = play_store_raw_df['installs'].str.replace(',','')
play_store_raw_df['installs'] = play_store_raw_df['installs'].str.replace('Free','0')

play_store_raw_df['price'] = play_store_raw_df['price'].str.replace('$','')
play_store_raw_df['price'] = play_store_raw_df['price'].str.replace('Everyone','')

play_store_raw_df[['reviews', 'installs', 'price']] = play_store_raw_df[['reviews', 'installs', 'price']].apply(pd.to_numeric,errors='ignore')
play_store_raw_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   app             10841 non-null  object 
 1   category        10841 non-null  object 
 2   rating          9367 non-null   float64
 3   reviews         10841 non-null  int64  
 4   installs        10841 non-null  int64  
 5   type            10840 non-null  object 
 6   price           10840 non-null  float64
 7   content_rating  10840 non-null  object 
dtypes: float64(2), int64(2), object(4)
memory usage: 677.7+ KB


Another potential issue to check for is duplicate apps. We can do this by using the df.value_counts() method.

In [8]:
play_store_raw_df.app.value_counts()

ROBLOX                                               9
CBS Sports App - Scores, News, Stats & Watch Live    8
Duolingo: Learn Languages Free                       7
Candy Crush Saga                                     7
ESPN                                                 7
                                                    ..
ASCP® CG Exam Flashcards 2018                        1
BMX Boy Bike Stunt Rider Game                        1
Android Auto - Maps, Media, Messaging & Voice        1
Ghost Hunting camera                                 1
Real Gangster Crime                                  1
Name: app, Length: 9660, dtype: int64

We can see that there are apps with duplicated rows in the dataset. For the purpose of this analysis, I want the latest updated version of the app. I can assume that the most updated version of the app data will be the one with the highest number of reviews.

In [10]:
play_store_df_filtered = play_store_raw_df.iloc[play_store_raw_df.groupby("app")["reviews"].idxmax()]
play_store_df_filtered.app.value_counts()

think Comics                                       1
eG Monitor                                         1
Minecraft                                          1
CW Nuclear                                         1
Fountain Live Wallpaper HD – Dubai Wallpaper 3D    1
                                                  ..
Daum Mail - Next Mail                              1
signály.cz                                         1
GO Security－AntiVirus, AppLock, Booster            1
Marked by King Bs                                  1
Real Gangster Crime                                1
Name: app, Length: 9660, dtype: int64

In [13]:
play_store_df = play_store_df_filtered.sort_index().reset_index(drop=True)
play_store_df['category'] = play_store_df.category.str.replace('_', ' ').str.title()
play_store_df

app             category  \
0        Photo Editor & Candy Camera & Grid & ScrapBook       Art And Design   
1     U Launcher Lite – FREE Live Cool Themes, Hide ...       Art And Design   
2                                 Sketch - Draw & Paint       Art And Design   
3                 Pixel Draw - Number Art Coloring Book       Art And Design   
4                            Paper flowers instructions       Art And Design   
...                                                 ...                  ...   
9655                                   Sya9a Maroc - FR               Family   
9656                   Fr. Mike Schmitz Audio Teachings               Family   
9657                             Parkinson Exercices FR              Medical   
9658                      The SCP Foundation DB fr nn5n  Books And Reference   
9659      iHoroscope - 2018 Daily Horoscope & Astrology            Lifestyle   

      rating  reviews  installs  type  price content_rating  
0        4.1      159     10000  Free    0.0       Everyone  
1        4.7    87510   5000000  Free    0.0       Everyone  
2        4.5   215644  50000000  Free    0.0           Teen  
3        4.3      967    100000  Free    0.0       Everyone  
4        4.4      167     50000  Free    0.0       Everyone  
...      ...      ...       ...   ...    ...            ...  
9655     4.5       38      5000  Free    0.0       Everyone  
9656     5.0        4       100  Free    0.0       Everyone  
9657     NaN        3      1000  Free    0.0       Everyone  
9658     4.5      114      1000  Free    0.0     Mature 17+  
9659     4.5   398307  10000000  Free    0.0       Everyone  

[9660 rows x 8 columns]

## Exploratory Analysis and Visualization

With our dataset cleaned to our satisfaction, we can perform some exploratory analysis and create some basic charts. First, we will use the `df.describe` method to analyze the numeric columns in our dataset and provide summary statistics.

In [12]:
play_store_df.describe()

rating       reviews      installs        price
count  8197.000000  9.660000e+03  9.660000e+03  9659.000000
mean      4.175076  2.167817e+05  7.797363e+06     1.097231
std       0.560670  1.831337e+06  5.376700e+07    16.851618
min       1.000000  0.000000e+00  0.000000e+00     0.000000
25%       4.000000  2.500000e+01  1.000000e+03     0.000000
50%       4.300000  9.690000e+02  1.000000e+05     0.000000
75%       4.500000  2.944925e+04  1.000000e+06     0.000000
max      19.000000  7.815831e+07  1.000000e+09   400.000000